test/Common.FunctionalTests/Inprocess/FrebTests.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.IIS.FunctionalTests.Utilities;
+using Microsoft.AspNetCore.Server.IntegrationTesting;
+using Microsoft.AspNetCore.Server.IntegrationTesting.IIS;
+using Microsoft.AspNetCore.Testing.xunit;
+using Microsoft.Extensions.Logging;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests
+{
+    [Collection(PublishedSitesCollection.Name)]
+    public class FrebTests : IISFunctionalTestBase
+    {
+        private readonly PublishedSitesFixture _fixture;
+
+        public FrebTests(PublishedSitesFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        [ConditionalTheory]
+        [RequiresIIS(IISCapability.FailedRequestTracingModule)]
+        [InlineData("/HelloWorld", "Verbose", "ANCM_EXECUTE_REQUEST_START", "ANCM_EXECUTE_REQUEST_COMPLETION", "ANCM_ASYNC_COMPLETION_START", "ANCM_ASYNC_COMPLETION_COMPLETION", "ANCM_MANAGED_REQUEST_COMPLETION")]
+        [InlineData("/Throw", "Verbose", "ANCM_EXECUTE_REQUEST_COMPLETION", "<Data Name=\"requestStatus\">2</Data>")] // Request status of 2 means request failed
+        public async Task CheckCommonFrebEvents(string requestPath, string verbosity, params string[] expectedEvents)
+        {
+            var parameters = _fixture.GetBaseDeploymentParameters(publish: true);
+            parameters.EnableFreb(verbosity);
+
+            var result = await DeployAsync(parameters);
+
+            await result.HttpClient.GetAsync(requestPath);
+
+            AssertFrebLogs(result, expectedEvents);
+        }
+
+        [ConditionalFact]
+        [RequiresIIS(IISCapability.FailedRequestTracingModule)]
+        public async Task CheckFrebDisconnect()
+        {
+            var parameters = _fixture.GetBaseDeploymentParameters(publish: true);
+            parameters.EnableFreb("Verbose");
+
+            var result = await DeployAsync(parameters);
+
+            using (var connection = new TestConnection(result.HttpClient.BaseAddress.Port))
+            {
+                await connection.Send(
```

Comment:
This would be racy, the request might not get to the handler before you disconnect.